LiteDB.Tests/Engine/Recursion_Tests.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using Xunit;
+
+namespace LiteDB.Tests.Engine;
+
+public class Recursion_Tests {
+    [Fact]
+    public void UpdateInFindAll()
+    {
+        Test(collection =>
```

Comment:
I'm not sure if this represents a [recursion](https://www.w3schools.com/c/c_functions_recursion.php#:~:text=Recursion%20is%20the%20technique%20of,is%20to%20experiment%20with%20it.), when do you say "recursion" you mean the idea of the method calling it self?

Reply:
I mean, previously the functions opened the database, did stuff and closed it. Like this:
```cs
OpenDatabase();
DoStuff();
CloseDatabase();
```
This means when you have an `Update()` in a `FindAll()`, it did something like this:
```cs
OpenDatabase();
Loop {
    OpenDatabase();
    DoStuff();
    CloseDatabase();
}
CloseDatabase();
```
Which sounds like recursion to me, I don't know if I mislabelled it.

Reply:
Oh, I know that as an iterator. But just names, thanks for clarifying it to me (: